### PR TITLE
MAINT Store zipped build folder as circleci artifact

### PR DESCRIPTION
**Pull request by hoodmane**
The goal is to make it easier to run the circleci build locally. With this we should be able to download a zip archive of the full build directory then serve it to localhost.

### CODE REVIEW

**Comment by rth** [comment]
At the point where we are, they will probably not mind if we store more things ). 

image

But we should probably try to optimize CircleCI more, as all of those numbers look quite high to me.

File: .circleci/config.yml [hunk 1]
```diff
@@ -97,6 +97,11 @@ jobs:
             rm -rf cpython/{build,downloads}
             rm -rf emsdk/emsdk/binaryen
 
+      - run:
+          name: Zip build directory
+          command: |
+            tar cjf pyodide-build.tar.bz2  build
```
Comment (rth) [inline .circleci/config.yml:103]: If I remember correctly bz2 takes a long time to compress, maybe better to use gz instead? Let's see what CI says.

Comment (hoodmane) [reply .circleci/config.yml:103]: 50 seconds on the build-packages job.

Comment (hoodmane) [reply .circleci/config.yml]: Indeed only 30 seconds to gzip. gzipped archive is exactly the same size as the bzipeped one.

**Pull request merged**